gamedat.cc

```diff
@@ -1065,7 +1065,7 @@ bool Game_window::save_gamedat_zip(
 		ofstream out;
 		char title[0x50];
 		memset(title, 0, 0x50);
-		std::strncpy(title, savename, 0x50);
+		memcpy(title, savename, 0x50);
```

Comment:
Once again, undefined behavior here whenever `strlen(savename) < 0x4f`. In this case, the only solution is doing something like `memcpy(title, savename, std::max(0x50, strlen(savename)))`.